third_party/nvidia/lib/Dialect/NVWS/Transforms/AssignStagePhase.cpp

```diff
@@ -218,24 +218,45 @@ template <class T> struct AssignStagePhase {
   StagePhase assignArefIndexInBlock(Block *block, StagePhase index) {
     for (auto &op : llvm::make_early_inc_range(*block)) {
       if (auto opT = isValidOp(&op)) {
-        ImplicitLocOpBuilder b(opT.getLoc(), opT);
-
-        auto nextStage = b.create<arith::AddIOp>(
-            index.stage, b.create<arith::ConstantIntOp>(1, 32));
+        ImplicitLocOpBuilder builder(opT.getLoc(), opT);
+        auto partitionIds = getPartitionIds(&op);
+        auto stageCluster = getStageCluster(&op);
+
+        auto createInto = [&](auto opTy, auto... args) {
+          using ty = decltype(opTy);
+          ty op = builder.create<ty>(std::forward<decltype(args)>(args)...);
+          if (partitionIds) {
+            // phase & stage computations are cloned to partition 0 as well
+            auto ids = *partitionIds;
+            ids.insert(0);
```

Comment:
I just noticed this, why is this necessary? This is incorrect for the inner loop in nested loops, since the inner loop is not copied into the default partition. But this codes add partition 0 to all stage / phase update in the inner loop